### PR TITLE
Add MemNode support

### PR DESCRIPTION
**Pull request by mtdudek**
This commit adds support for PeakRDL MemNode objects. Register arrays inside MemNode are not being unrolled to keep memory description brief.
Memories can hold hundreds or even more virtual registers and unrolling all of them is not beneficial and mudders the overall picture.

Solves #12

### CODE REVIEW

File: src/peakrdl_markdown/exporter.py [hunk 1]
```diff
@@ -194,9 +208,10 @@ def _add_addrmap_regfile(
         """
         members: List[MarkdownExporter.GenStageOutput] = []
         member_gen: str = ""
-        for child in node.children(unroll=True, skip_not_present=False):
-            if isinstance(child, (AddrmapNode, RegfileNode)):
-                output = self._add_addrmap_regfile(child, msg, depth - 1)
+        not_memory = not isinstance(node, MemNode)
```
Comment (MarekPikula) [inline src/peakrdl_markdown/exporter.py:211]: Please add a comment here about why we don't want to unroll MemNodes.

File: src/peakrdl_markdown/exporter.py [hunk 2]
```diff
@@ -95,10 +107,12 @@ def _addrnode_header(self, node: AddressableNode, heading_level: int) -> str:
             node_type_name = "address map"
         elif isinstance(node, RegfileNode):
             node_type_name = "register file"
+        elif isinstance(node, MemNode):
+            node_type_name = "memory"
         elif isinstance(node, RegNode):
             node_type_name = "register"
         else:
-            node_type_name = "addressable node"
+            raise ValueError(f"Unknown AddressableNode type {type(node)}")
```
Comment (MarekPikula) [inline src/peakrdl_markdown/exporter.py:115]: Why did you change it to hard fail here? I'd prefer more "graceful" fail, like warning on console and ignoring the field in the output (like it was before with MemNode).

**Review by MarekPikula — CHANGES_REQUESTED** [review]
Thank you for your contribution!

**Comment by MarekPikula** [comment]
Two more things:

1. Please rebase onto the latest `main`. I did some tweaks in Python dependencies (included 3.6 support) and in CI.
2. Add the information about added MemNode support to changelog.

**Comment by mtdudek** [comment]
I've updated CHANGELOG and rebased onto the latest `main`. I also addressed your commends.

File: src/peakrdl_markdown/exporter.py (hunk 1 above)
Comment (mtdudek) [reply src/peakrdl_markdown/exporter.py:216]: Added comment about memories

**Review by MarekPikula — APPROVED** [review]

**Review by MarekPikula — CHANGES_REQUESTED** [review]
Sorry, I noticed that there are small style issues in CI. I'll fix those.

**Comment by MarekPikula** [comment]
Thanks again for the contribution. It's released now as 1.0.1.
